### DISCUSSION TITLE
What method computes nodal forces / moments in beam

### DISCUSSION THREAD

**Discussion by andrey1887** [idaholab/moose#26145] [Q&A Modules: Solid mechanics] [answered]
Hello. I am looking for method, which computes force / moment values in the nodes of line element. `ComputeBeamResultants::computeQpProperties()` gives their values in q-points, but what about nodal values. Please help me to find it.

**Comment by GiudGiud** [comment]
Hello

What are force and moment stored as? Material properties? 

Or are you computing them yourself from other quantities? (And in this case, what are they stored as?)

Guillaume

**Reply by andrey1887** [reply]
Thank you, @GiudGiud . Force and moment in q-points stored as material properties 
![image](https://github.com/idaholab/moose/assets/114737834/00063316-5dd1-4249-9220-373739dae26c)
It computed from strains in ComputeIncrementalBeamStrain::computeQpStrain()
But it all about qp-Forces

i can't find where nodal forces are computed, maybe in StressDivergenceBeam ?

**Reply by GiudGiud** [reply]
I don't know either. 
@cbolisetti do we compute them?

@hugary1995 do we have patch reconstruction to compute nodal material properties for beam elements?

**Reply by hugary1995** [reply, marked as answer]
The tagged residual vector should be the reaction forces and moments.

**Reply by andrey1887** [reply]
Thank you @hugary1995, @GiudGiud.
